server/src/completions.ts

```diff
@@ -364,68 +401,115 @@ function getFieldsFromCurrentBlock(
     if (key === block.end.line) {
       break
     }
-    if (key !== position.line) {
+    if (!position) {
+      suggestions.push(item.replace(/ .*/, ''))
+    } else if (key !== position.line) {
       suggestions.push(item.replace(/ .*/, ''))
     }
   }
+  return suggestions
+}
 
-  const currentLine = lines[position.line]
-  if (currentLine.includes('fields') && currentLine.includes('references')) {
-    const otherContext = context === 'references' ? 'fields' : 'references'
-    // remove all fields that might be inside other attribute
-    // e.g. if inside 'references' context, fields from 'fields' context are not correct suggestions
-    const startOfOtherContext = currentLine.indexOf(otherContext + ': [')
-    let end: number = startOfOtherContext
-    for (end; end < currentLine.length; end++) {
-      if (currentLine.charAt(end) === ']') {
-        break
-      }
+function getSuggestionsForRelationDirective(
+  wordsBeforePosition: string[],
+  currentLineUntrimmed: string,
+  lines: string[],
+  block: Block,
+  position: Position,
+): CompletionList | undefined {
+  const wordBeforePosition = wordsBeforePosition[wordsBeforePosition.length - 1]
+  const stringTillPosition = currentLineUntrimmed
+    .slice(0, position.character)
+    .trim()
+
+  if (wordBeforePosition.includes('@relation')) {
+    return {
+      items: toCompletionItems(
+        ['references: []', 'fields: []', '""'],
```

Comment:
what's it doing here? Are these the options you can choose from?

Reply:
exactly. So if you are inside `@relation(_here_)`, it will suggest `references: []`, `fields: []` and `""`